internal/schema/connection.go

```diff
@@ -314,13 +322,22 @@ func GetEsConnectionSchema(keyName string, isProviderConfiguration bool) *schema
 }
 
 func GetKibanaConnectionSchema() *schema.Schema {
+	withEnvDefault := func(key string, dv interface{}) schema.SchemaDefaultFunc { return nil }
 	return &schema.Schema{
 		Description: "Kibana connection configuration block.",
 		Type:        schema.TypeList,
 		MaxItems:    1,
 		Optional:    true,
 		Elem: &schema.Resource{
 			Schema: map[string]*schema.Schema{
+				"api_key": {
+					Description:   "API Key to use for authentication to Kibana",
+					Type:          schema.TypeString,
+					Optional:      true,
+					Sensitive:     true,
+					DefaultFunc:   withEnvDefault("KIBANA_API_KEY", nil),
```

Comment:
```suggestion
```

Env variables are being read in the config code. This has a different priority ordering to the config code (env variables don't override hardcoded attributes) as well so can be confusing.

Reply:
> AFAICT mirrors ES schema

Yea ok. Let's leave it like this :)